h/tasks/indexer.py

```diff
@@ -6,13 +8,23 @@
 log = get_task_logger(__name__)
 
 
-@celery.task
+# See: https://docs.celeryproject.org/en/stable/userguide/tasks.html#automatic-retry-for-known-exceptions
+class _BaseTaskWithRetry(Task):
+    autoretry_for = (Exception,)
+    retry_kwargs = {"max_retries": 5}
+    # Add exponential back-off
+    retry_backoff = True
+    # Shuffle the times a bit to prevent the thundering herd problem
+    retry_jitter = True
```

Comment:
So I think this will retry after 1, 2, 4, 8 and then 16 second delays (plus or minus random jitter)? So 31 seconds in Celery plus random jitter time plus the time actually spent on up to 6 processing attempts. So if the periodic tasks ignores any annotation events less than one minute old, that should be plenty safe that the periodic task and these tasks are out of each other's way.

Reply:
Hmm reading this again, I don't think this is necessarily what I thought:

 * https://docs.celeryproject.org/en/stable/userguide/tasks.html#Task.retry_jitter

> A boolean. Jitter is used to introduce randomness into exponential backoff delays, to prevent all tasks in the queue from being executed simultaneously. If this option is set to True, the delay value calculated by retry_backoff is treated as a maximum, and the actual delay value will be a random number between zero and that maximum. By default, this option is set to True.

This implies there that jitter will randomly choose between 0 and 1, 2, 4, 8 etc. So enabling jitter will on average half the time taken, so about 16s. 

We could set retry back-off to 2 to counteract this so the delays start at 2, 4, etc?

Reply:
Yeah, I think let's do that. I think chilling it out a little bit is a good idea. I don't like the idea that it can retry immediately (_the actual delay value will be a random number between zero and that maximum_) but it sounds like, with jitter, that potential is unavoidable, but we can at least add the multiplication factor to make it less common.